common/multigrid/amgx_pgm_kernels.hpp.inc

```diff
@@ -0,0 +1,341 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2021, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+namespace kernel {
+
+
+template <typename IndexType>
+__global__ __launch_bounds__(default_block_size) void replace_kernel(
+    size_type size, const IndexType *__restrict__ source,
+    IndexType *__restrict__ result)
+{
+    auto tidx = thread::get_thread_id_flat();
+    if (tidx >= size) {
+        return;
+    }
+
+    result[tidx] = source[tidx] == -1;
+}
+
+
+template <typename IndexType>
+__global__ __launch_bounds__(default_block_size) void match_edge_kernel(
+    size_type num, const IndexType *__restrict__ strongest_neighbor_vals,
+    IndexType *__restrict__ agg_vals)
+{
+    auto tidx = thread::get_thread_id_flat<IndexType>();
+    if (tidx >= num) {
+        return;
+    }
+    if (agg_vals[tidx] != -1) {
+        return;
+    }
+    auto neighbor = strongest_neighbor_vals[tidx];
+    if (neighbor != -1 && strongest_neighbor_vals[neighbor] == tidx) {
+        // Use the smaller index as agg point
+        auto group = min(tidx, neighbor);
+        agg_vals[tidx] = group;
+        agg_vals[neighbor] = group;
+    }
+}
+
+
+template <typename IndexType>
+__global__ __launch_bounds__(default_block_size) void activate_kernel(
```

Comment:
Isn't this exactly the same as `replace_kernel`?

Reply:
Yes, it is the same. I forgot to delete it